backend/ethereum/channel/conclude.go

```diff
@@ -69,24 +79,44 @@ func (a *Adjudicator) ensureConcluded(ctx context.Context, req channel.Adjudicat
 
 	// No conclude event found in the past, send transaction.
 	if req.Tx.IsFinal {
-		err = errors.WithMessage(a.callConcludeFinal(ctx, req), "calling concludeFinal")
+		err = cherrors.CheckIsChainNotReachableError(a.callConcludeFinal(ctx, req))
+		err = errors.WithMessage(err, "calling concludeFinal")
 	} else {
-		err = errors.WithMessage(a.callConclude(ctx, req, subStates), "calling conclude")
+		err = cherrors.CheckIsChainNotReachableError(a.callConclude(ctx, req, subStates))
+		err = errors.WithMessage(err, "calling conclude")
 	}
 	if IsErrTxFailed(err) {
 		a.log.Warn("Calling conclude(Final) failed, waiting for event anyways...")
 	} else if err != nil {
 		return err
 	}
 
-	select {
-	case <-events:
-		return nil
-	case <-ctx.Done():
-		return errors.Wrap(ctx.Err(), "context cancelled")
-	case err = <-sub.Err():
-		err = cherrors.CheckIsChainNotReachableError(err)
-		return errors.Wrap(err, "subscription error")
+	for {
+		select {
+		case _e := <-events:
+			e := _e.Data.(*adjudicator.AdjudicatorChannelUpdate)
+			if e.Phase == phaseConcluded {
+				return nil
+			}
+		case <-ctx.Done():
+			return errors.Wrap(ctx.Err(), "context cancelled")
+		case err = <-subErr:
+			if err != nil {
+				return errors.WithMessage(err, "subscription error")
+			}
+			return errors.New("subscription closed")
+		}
+	}
+}
+
+func updateEventType(channelID [32]byte) subscription.EventFactory {
+	return func() *subscription.Event {
+		return &subscription.Event{
+			Name: "ChannelUpdate",
+			Data: new(adjudicator.AdjudicatorChannelUpdate),
+			// In the best case we could already filter for 'Concluded' phase only here.
+			Filter: [][]interface{}{{channelID}},
```

Comment:
Can we filter for `phase == Concluded`?

Reply:
Sadly not. We could add an extended filtering logic to the `EventSub`, i will record it on an issue.

Reply:
Why not? Could it be because we are [not indexing the phase](https://github.com/hyperledger-labs/perun-eth-contracts/blob/27efac547588bc841151701000e289becf989739/contracts/Adjudicator.sol#L64)?